### PR TITLE
[#18600] feat: implement set bio screen

### LINKED ISSUE #18453: Edit profile - Change Accent colour
### User Story
As a user
I want to be able to change my accent colour.

This is the edit accent colour flow:

https://www.figma.com/file/PiN0uwRgZj4L9U4N7xO7Xb/Profile-for-Mobile?type=design&node-id=4528-314950&mode=design&t=VygLdQE5rvI3yTdG-4

### LINKED ISSUE #18600: Edit profile - Edit bio
### User Story

As a user
I want to be able to edit my bio.

This is the edit bio flow : 

https://www.figma.com/file/PiN0uwRgZj4L9U4N7xO7Xb/Profile-for-Mobile?type=design&node-id=4409-370351&mode=design&t=BaUyPFPCfN6hynui-4

### PR DESCRIPTION
**Pull request by mohsen-ghafouri**
fixes #18600

### Summary
Implement set profile bio

### Steps to test

1. Open Status
2. Click on profile avatar (Settings screen)
3. Click on Edit Profile
4. Click on Bio
5. Enter new nio text
6. Click on Save bio

### Result

https://github.com/status-im/status-mobile/assets/71308738/b5e3a5fb-0721-43f0-ba66-b0119468956b

status: ready

### CODE REVIEW

File: src/status_im/contexts/profile/edit/bio/events_test.cljs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+(deftest edit-bio-test
+  (let [new-bio  "New Bio text"
+        cofx     {:db {:profile/profile {:bio "Old Bio text"}}}
+        expected {:db {:profile/profile {:bio new-bio}}
+                  :fx [[:json-rpc/call
+                        [{:method     "wakuext_setBio"
+                          :params     [new-bio]
+                          :on-success [:profile/edit-profile-bio-success]}]]]}]
+    (is (match? expected
+                (sut/edit-profile-bio cofx [new-bio])))))
```
Comment (smohamedjavid) [inline src/status_im/contexts/profile/edit/bio/events_test.cljs:15]: ❤️

**Review by smohamedjavid — APPROVED** [review]
Nice work! @mohsen-ghafouri 🙌 🚀

**Review by jo-mut — APPROVED** [review]

**Comment by mariia-skrypnyk** [comment]
Hi @mohsen-ghafouri !
Thanks for your PR! Curious to test it!

I have **several questions** to ask:

**1)** First question is regarding screen animation. 
I feel like the first thing that should happen is Bio screen opening and then the keyboard (if that's a plan to start typing). 
But now both these actions take place at the same time. And it gives me the filling that something wrong is going on here.
Let me show you: 

https://github.com/status-im/status-mobile/assets/147824399/ed44856b-1b43-4e4b-9b47-4a07b4c5824e

Does it works as designed or it has an incorrect behaviour?

**2)** Should Bio be opened using animation not from the bottom to the top but sliding from the right to the left? (sorry for my nub explanation manner:) You also can see on the video above how I open another screens from the Profile to compare the consistency of this animation. 
Let me know is it correct too.

**Comment by mohsen-ghafouri** [comment]
Thank you @mariia-skrypnyk for testing this PR. 
for point 1 according to design https://www.figma.com/file/PiN0uwRgZj4L9U4N7xO7Xb/Profile-for-Mobile?type=design&node-id=4409-370351&mode=design&t=Kkft99yoNyEfuf7v-4, the screen will open with opened keyboard. but in design review step they can give us more feedback. 

for point 2 we had this comment from design team https://github.com/status-im/status-mobile/pull/18517#pullrequestreview-1865817703, so all screen with back button should open from right side.

**Comment by mariia-skrypnyk** [comment]
> Thank you @mariia-skrypnyk for testing this PR. for point 1 according to design https://www.figma.com/file/PiN0uwRgZj4L9U4N7xO7Xb/Profile-for-Mobile?type=design&node-id=4409-370351&mode=design&t=Kkft99yoNyEfuf7v-4, the screen will open with opened keyboard. but in design review step they can give us more feedback.
> 
> for point 2 we had this comment from design team [#18517 (review)](https://github.com/status-im/status-mobile/pull/18517#pullrequestreview-1865817703), so all screen with back button should open from right side.

Thanks for requirements and your feedback!

Have a small issue:

**ISSUE 1: no placeholder in Bio field** 
Design vs Current
Screenshot 2024-02-13 at 17 59 44

**Comment by mariia-skrypnyk** [comment]
@mohsen-ghafouri please, let me know when it will be ready for me to test!

**Comment by mohsen-ghafouri** [comment]
@mariia-skrypnyk resolved

**Comment by mariia-skrypnyk** [comment]
@mohsen-ghafouri 
Thanks for your fix. Placeholder looks great now.
I've finished my QA part and going to move your PR to the Design Review for @Francesca-G approve.

**Review by Francesca-G — APPROVED** [review]
Nice work ✨

Adding the follow up required label to address some minor issues about alignment and button state, as commented [here](https://www.figma.com/file/OU646EHTHdj45z55dhxRDg/Review-Profile-%2F-Mobile?type=design&node-id=13%3A12427&mode=design&t=zxBkarfrHI4oSH31-1)

**Comment by mariia-skrypnyk** [comment]
> Nice work ✨
> 
> Adding the follow up required label to address some minor issues about alignment and button state, as commented [here](https://www.figma.com/file/OU646EHTHdj45z55dhxRDg/Review-Profile-%2F-Mobile?type=design&node-id=13%3A12427&mode=design&t=zxBkarfrHI4oSH31-1)

@mohsen-ghafouri can you please create followup on Francesca's feedback?

**Pull request merged**